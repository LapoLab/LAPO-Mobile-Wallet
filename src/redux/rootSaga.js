import { all } from 'redux-saga/effects';
import commonsSagas from '../commons/sagas';
import recoveryWordsSagas from '../modules/RecoveryWords/sagas';
import sendSagas from '../modules/Send/sagas';
import paperWordsConfirm from '../modules/PaperWordsConfirm/sagas';
import dashboard from '../modules/Dashboard/sagas';
import settings from '../modules/Settings/sagas';
import paperWords from '../modules/PaperWords/sagas';
import transactionsDetails from '../modules/TransactionDetails/sagas';
import chooseContact from '../modules/ChooseContact/sagas';
import splashScreen from '../modules/SplashScreen/sagas';
import switchContacts from '../modules/SwitchContacts/sagas';
import receive from '../modules/Receive/sagas';

// eslint-disable-next-line
const getListeners = (...args) => args.reduce((acc, nextArg) => [...acc, ...Object.values(nextArg).map(func => func())], []);

export default function* rootSaga() {
  yield all(
    getListeners(
      commonsSagas,
      recoveryWordsSagas,
      sendSagas,
      paperWordsConfirm,
      dashboard,
      settings,
      paperWords,
      transactionsDetails,
      chooseContact,
      splashScreen,
      switchContacts,
      receive,
    ),
  );
}
