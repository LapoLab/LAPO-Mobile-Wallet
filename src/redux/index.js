import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import rootReducer from './rootReducer';
import rootSagas from './rootSaga';

const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
  blacklist: [
    'modals',
    'dashboard',
    'paperWords',
    'chooseContact',
    'send',
    'transactionsDetails',
    'alert',
    'pinsRecovery',
  ],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const sagaMiddleware = createSagaMiddleware();
// eslint-disable-next-line no-undef
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSagas);

export const persistor = persistStore(store);
