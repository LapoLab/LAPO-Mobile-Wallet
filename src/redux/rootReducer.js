import { combineReducers } from 'redux';

import wallet from '../commons/reducers/wallet';
import pins from '../commons/reducers/pins';
import pinsRecovery from '../commons/reducers/pinsRecovery';
import modals from '../commons/reducers/modals';
import settings from '../commons/reducers/settings';
import notify from '../modules/Notify/reducers';
import paperConfirm from '../modules/PaperWordsConfirm/reducer';
import dashboard from '../modules/Dashboard/reducers';
import paperWords from '../modules/PaperWords/reducers';
import chooseContact from '../modules/ChooseContact/reducers';
import send from '../modules/Send/reducers';
import transactionsDetails from '../modules/TransactionDetails/reducer';
import alert from '../modules/Alert/reducers';

export default combineReducers({
  wallet,
  pins,
  pinsRecovery,
  notify,
  modals,
  settings,
  paperConfirm,
  dashboard,
  paperWords,
  chooseContact,
  send,
  transactionsDetails,
  alert,
});
