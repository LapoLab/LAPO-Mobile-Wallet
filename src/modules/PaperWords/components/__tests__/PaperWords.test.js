// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../redux';
import { PaperWords } from '../index';

describe('PaperWords page', () => {
  test('renders snapshots', () => {
    const navigation = {
      addListener: jest.fn(),
      dispatch: jest.fn(),
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <PaperWords
            navigation={navigation}
            setConfirmPins={jest.fn()}
            setPaperWords={jest.fn()}
            setPins={jest.fn()}
            generatePhrase={jest.fn()}
            tempWords={['123', '234', '345']}
            share12Words={jest.fn()}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
