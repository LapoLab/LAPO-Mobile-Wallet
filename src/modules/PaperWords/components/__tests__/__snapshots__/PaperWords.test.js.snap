// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaperWords page renders snapshots 1`] = `
<BVLinearGradient
  colors={
    Array [
      -12483651,
      -16172276,
    ]
  }
  endPoint={
    Object {
      "x": 1.2,
      "y": 1.2,
    }
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  startPoint={
    Object {
      "x": -0.1,
      "y": -0.1,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "space-between",
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/back.png",
            }
          }
          style={
            Object {
              "marginLeft": 12,
              "resizeMode": "contain",
              "width": 12,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "white",
            "fontFamily": "Lato-Regular",
            "fontSize": 22,
            "marginLeft": 30,
            "paddingLeft": 10,
            "textAlign": "center",
          }
        }
      >
        Paper key
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "marginRight": 12,
            }
          }
        >
          Save to pdf
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 50,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/indicator3.png",
          }
        }
        style={
          Object {
            "height": 20,
            "resizeMode": "contain",
          }
        }
      />
    </View>
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 180,
              "justifyContent": "center",
              "paddingHorizontal": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Lato-Light",
                "fontSize": 20,
                "marginTop": 20,
                "textAlign": "center",
              }
            }
          >
            Your paper key is the only way to restore your LAPO Mobile wallet if your phone is lost, stolen, broken or upgraded.
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Lato-Light",
                "fontSize": 20,
                "marginTop": 20,
                "textAlign": "center",
              }
            }
          >
            We will show you a list of words to write down in this exactly order on a piece of paper and keep safe.
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "height": 200,
              "marginTop": 10,
              "paddingHorizontal": 20,
              "paddingVertical": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0C3841",
                  "borderRadius": 8,
                  "justifyContent": "center",
                  "marginVertical": 5,
                  "paddingHorizontal": 10,
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Lato-Bold",
                    "fontSize": 14,
                  }
                }
              >
                1.  
                123
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0C3841",
                  "borderRadius": 8,
                  "justifyContent": "center",
                  "marginVertical": 5,
                  "paddingHorizontal": 10,
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Lato-Bold",
                    "fontSize": 14,
                  }
                }
              >
                2.  
                234
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0C3841",
                  "borderRadius": 8,
                  "justifyContent": "center",
                  "marginVertical": 5,
                  "paddingHorizontal": 10,
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Lato-Bold",
                    "fontSize": 14,
                  }
                }
              >
                3.  
                345
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "height": 100,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#4183BD",
              "borderColor": "rgb(42, 101, 117)",
              "borderRadius": 74,
              "borderWidth": 8,
              "height": 64,
              "justifyContent": "center",
              "marginHorizontal": 25,
              "opacity": 1,
              "width": "80%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Lato-Regular",
                "fontSize": 18,
              }
            }
          >
            OK, I NOTED IT
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
